### PR TITLE
Add support for multi react instance apps

### PR DESCRIPTION
**Pull request by Szymon20000**
# Why

Some apps contain several react instances and some unimodules need additional information from users. In order to make possible for unimodule instances to have different configurations from each other I want to assign them unique ids.

# How

# Test Plan

### CODE REVIEW

**Comment by ide** [comment]
I'm not sure we want to take on the complexity from this. There are potentially some security considerations and bundle splitting might be able to address a similar use case. For a change this significant we need to consider it thoroughly and be able to justify the maintenance cost (building features is easy, maintaining them is hard).

File: templates/expo-template-bare-typescript/android/app/src/main/java/com/helloworld/MainApplication.java [hunk 1]
```diff
@@ -27,7 +27,8 @@
 public class MainApplication extends Application implements ReactApplication {
   private final ReactModuleRegistryProvider mModuleRegistryProvider = new ReactModuleRegistryProvider(
     new BasePackageList().getPackageList(),
-    Arrays.<SingletonModule>asList()
+    Arrays.<SingletonModule>asList(),
+    "INJECT_APP_ID_HERE"
```
Comment (sjchmiela) [inline templates/expo-template-bare-typescript/android/app/src/main/java/com/helloworld/MainApplication.java]: If we moved this property to inside `mReactNativeHost` we could use `getJSMainModuleName` here?

File: templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.java [hunk 2]
```diff
@@ -27,7 +27,8 @@
 public class MainApplication extends Application implements ReactApplication {
   private final ReactModuleRegistryProvider mModuleRegistryProvider = new ReactModuleRegistryProvider(
     new BasePackageList().getPackageList(),
-    Arrays.<SingletonModule>asList()
+    Arrays.<SingletonModule>asList(),
+    "INJECT_APP_ID_HERE"
```
Comment (sjchmiela) [inline templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.java]: If we moved this property to inside `mReactNativeHost` we could use `getJSMainModuleName` here?

File: packages/@unimodules/core/android/src/main/java/org/unimodules/core/ModuleRegistry.java [hunk 3]
```diff
@@ -20,11 +20,17 @@
   private final Map<String, SingletonModule> mSingletonModulesMap = new HashMap<>();
   private final List<WeakReference<RegistryLifecycleListener>> mExtraRegistryLifecycleListeners = new ArrayList<>();
 
+  private String mAppId = "";
```
Comment (sjchmiela) [inline packages/@unimodules/core/android/src/main/java/org/unimodules/core/ModuleRegistry.java]:
```suggestion
  private String mAppId;
```

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ExpoModuleRegistryAdapter.java [hunk 4]
```diff
@@ -29,6 +29,9 @@
 import versioned.host.exp.exponent.modules.universal.sensors.ScopedRotationVectorSensorService;
 
 public class ExpoModuleRegistryAdapter extends ModuleRegistryAdapter implements ScopedModuleRegistryAdapter {
+
+  private String mAppId;
+
```
Comment (sjchmiela) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ExpoModuleRegistryAdapter.java]: Removable block

File: android/expoview/src/main/java/versioned/host/exp/exponent/ExponentPackage.java [hunk 5]
```diff
@@ -223,6 +226,16 @@ private void addViewManagersFromPackages(ReactApplicationContext reactContext,
...
+    return new ExpoModuleRegistryAdapter(new ReactModuleRegistryProvider(packages, singletonModules, mExperienceId));
+  }
+
+  private String getExperienceIdFormManifest(JSONObject manifest) {
+    try {
+      ExperienceId experienceId = ExperienceId.create(manifest.getString(ExponentManifest.MANIFEST_ID_KEY));
+      return experienceId.get();
+    } catch (JSONException e) {
+      e.printStackTrace();
+      return null;
```
Comment (sjchmiela) [inline android/expoview/src/main/java/versioned/host/exp/exponent/ExponentPackage.java]: Is the `null` safe to use? We should make a conscious decision about nullability of `appId`.

**Comment by sjchmiela** [comment]
We no longer feel this change is needed, but if it ever arises again we'll make sure to return to this PR. 👍

**Pull request closed without merging**